lib/Analysis/Utility.cpp

```diff
@@ -797,8 +797,7 @@ bool matchMFMAAndDotOperandShuffleCase(RankedTensorType srcTy,
   // Currently supporting 32x32 and 16x16 FP8 MFMA -> dot operand case
   return dotOperandLayout.getParent() == mfmaLayout &&
          dotOperandLayout.getOpIdx() == 0 && mfmaLayout.getIsTransposed() &&
-         dotOperandLayout.getKWidth() == 8 &&
-         getContigPerThread(mfmaLayout)[1] == 4 &&
+         dotOperandLayout.getKWidth() == 8 && !mfmaLayout.getIsTransposed() &&
```

Comment:
I believe this is what the previous code wanted to do querying `getContigPerThread`, could you confirm?
The thing is that `getContigPerThread` may return different (correct) numbers depending on broadcasting, so better not relying on those.